src/main/java/org/openjfx/gradle/JavaFXPlatform.java

```
@@ -0,0 +1,30 @@
+package org.openjfx.gradle;
+
+import com.google.gradle.osdetector.OsDetector;
+import org.gradle.api.Project;
+
+public enum JavaFXPlatform {
+
+    LINUX("linux"),
+    WINDOWS("win"),
+    OSX("mac");
+
+    private String classifier;
+
+    JavaFXPlatform( String classifier ) {
+        this.classifier = classifier;
+    }
+
+    public String getClassifier() {
+        return classifier;
+    }
+
+    public static JavaFXPlatform detect(Project project) {
+        String os = project.getExtensions().getByType(OsDetector.class).getOs();
+        switch (os) {
+            case "osx"    : return JavaFXPlatform.OSX;
+            case "windows": return JavaFXPlatform.WINDOWS;
+            default       : return JavaFXPlatform.LINUX;
```

I think unsupported platforms should be dealt with in a different way then depending on the linux distribution of JavaFX. Here is a [list of the platform names](https://github.com/trustin/os-maven-plugin#property-osdetectedname) that can possibly be detected by the osdetector plugin.

Some of the options we have for handling unsupported platforms are:

1. ignore the fact and use linux as default (this pull request)
2. don't use linux as default and silently skip adding the javafx dependencies to the project
3. print a warning that an unsupported platform is being used
4. throw an exception saying that it's not possible to use the plugin on an unsupported platform

---

Option 4 seems to be the best IMO. I updated the PR to do just that.

---

Please merge at your convenience